android/mobile/app/src/main/java/com/ttd/sso/sample/MainActivity.kt

```diff
@@ -0,0 +1,119 @@
+package com.ttd.sso.sample
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import com.ttd.sso.sample.ui.AuthScreenAction.CheckSignIn
+import com.ttd.sso.sample.ui.AuthScreenAction.SignIn
+import com.ttd.sso.sample.ui.AuthScreenAction.SignOut
+import com.ttd.sso.sample.ui.AuthScreenState.ErrorState
+import com.ttd.sso.sample.ui.AuthScreenState.SignedInState
+import com.ttd.sso.sample.ui.AuthScreenState.SignedOutState
+import com.ttd.sso.sample.ui.AuthScreenViewModel
+import com.ttd.sso.sample.ui.AuthScreenViewModelFactory
+
+class MainActivity : ComponentActivity() {
+
+    private val viewModel: AuthScreenViewModel by viewModels {
+        AuthScreenViewModelFactory()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            MaterialTheme {
+                AuthScreen(viewModel = viewModel)
+            }
+        }
+    }
+
+    override fun onPostCreate(savedInstanceState: Bundle?) {
+        super.onPostCreate(savedInstanceState)
+        handleIntent(intent)
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+        handleIntent(intent)
+    }
+
+    private fun handleIntent(intent: Intent) {
+        viewModel.processAction(CheckSignIn(intent))
+    }
+}
+
+@Composable
+fun AuthScreen(viewModel: AuthScreenViewModel, modifier: Modifier = Modifier) {
+    val context = LocalContext.current
+    val viewState by viewModel.viewState.collectAsState()
+
+    Box(modifier = modifier) {
+        Column(
```

Comment:
What's the Box purpose? it seems it's only the whole column in it? If so, using the Column directly would be enough?

However if you add a loading state, it would probably need to stay.